### PR TITLE
chore: add structured task log data to API

### PR DESCRIPTION
**Pull request by maxrussell**
## Description

This is to address [DET-7103] by adding task logs' fields to the API rather than merely templating out a string. This should allow the frontend to update how this is displayed without updating the backend.

## Commentary

I added the fields at the top level of `TaskLogsResponse` because it seemed to better fit the other types in the API. Let me know if this seems suboptimal.

[DET-7103]: https://determinedai.atlassian.net/browse/DET-7103?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by hkang1 — COMMENTED** [review]
Just one question, but the proposal looks good and matches with what Web expects and uses.

File: webui/react/src/services/api-ts-sdk/api.ts [hunk 1]
```diff
@@ -5506,6 +5506,54 @@ export interface V1TaskLogsResponse {
...
+     * @type {number}
+     * @memberof V1TaskLogsResponse
+     */
+    rankId?: number;
+    /**
+     * The text of the log entry.
+     * @type {string}
+     * @memberof V1TaskLogsResponse
+     */
+    log: string;
```
[the remark this answers is not in the document]
Comment (maxrussell) [reply webui/react/src/services/api-ts-sdk/api.ts:5544]: If I'm reading things correctly, this should just be the raw message that comes back. This value is used (as a suffix) to generate the `message` field that was and still is included in the response [here](https://github.com/determined-ai/determined/blob/master/master/pkg/model/task.go#L325).

**Review by stoksc — COMMENTED** [review]
lgtm. three points
1. can we mark `message` as deprecated?
2. can we make a ticket to move the CLI off the message field?
3. @hkang1 when do you think the webui would want to move over to this so we can remove `message`?

**Comment by maxrussell** [comment]
@stoksc

1. [Done!](https://github.com/determined-ai/determined/pull/4270/commits/4a4b1cf33fb67503ecda440b515b0381696b8a7f)
2. [Done!](https://determinedai.atlassian.net/browse/DET-7561)

**Comment by hkang1** [comment]
> lgtm. three points
> 
> 1. can we mark `message` as deprecated?
> 2. can we make a ticket to move the CLI off the message field?
> 3. @hkang1 when do you think the webui would want to move over to this so we can remove `message`?

I can get the web changes in by tomorrow

**Review by stoksc — APPROVED** [review]
sick

**Pull request closed without merging**